### PR TITLE
Prepare for v3.1.4 release

### PR DESCRIPTION
**Pull request by jsquyres**
@bwbarrett Note that I updated R for all the .so versions. See the commit message.

Refs #5990.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -55,6 +55,43 @@ included in the vX.Y.Z section and be denoted as:
...
 
+3.1.4 -- February, 2019
+-----------------------
+
+- Fix compile error when configured with --enable-mpi-java and
+  --with-devel-headers.  Thanks to @g-raffy for reporting the issue
+  (** also appeared: v3.0.4).
+- Only use hugepages with appropriate permissions.  Thanks to Hunter
+  Easterday for the fix.
+- Fix possible floating point rounding and division issues in OMPIO.
```
Comment (ax3l) [inline NEWS]: Those are not floating point rounding issues (this is just the implementation detail causing them), but they are
- crashes and
- data corruption

described in #6285 with currently four PRs for fixes (see PR description for full list).

Comment (jsquyres) [reply NEWS]: Done.

File: NEWS [hunk 2]
```diff
@@ -55,6 +55,44 @@ included in the vX.Y.Z section and be denoted as:
...
+3.1.4 -- February, 2019
+-----------------------
+
+- Fix compile error when configured with --enable-mpi-java and
+  --with-devel-headers.  Thanks to @g-raffy for reporting the issue
+  (** also appeared: v3.0.4).
+- Only use hugepages with appropriate permissions.  Thanks to Hunter
+  Easterday for the fix.
+- Fix possible floating point rounding and division issues in OMPIO
+  which led to crashes and/or data corruption with with very large
```
Comment (ax3l) [inline NEWS]: Sorry to bump on this one more time: as we just found out in https://github.com/open-mpi/ompi/issues/6285#issuecomment-466645046, the data corruption fixes are not yet backported, only the crashes.
```suggestion
  which led to crashes with large
```

(~~very~~: magic ranges every 400 MB is not too large :) )

Comment (jsquyres) [reply NEWS]: We'll get the data corruption fix in.  I'm not worried about the relatively meaning of "very".  😄

**Comment by jsquyres** [comment]
@bwbarrett The NEWS in this PR assumes we get a fix for #6436.

File: NEWS (hunk 2 above)
Comment (ax3l) [reply NEWS]: haha, yes that's ok with me :)
Thanks for pushing the DDT backports!

**Comment by jsquyres** [comment]
False CI failure (infamous Jenkins-cant-checkout issue).

bot:ompi:retest

**Comment by jsquyres** [comment]
This is a text file update; perhaps the Mellanox CI failure was transient...

bot:mellanox:retest

**Comment by jsquyres** [comment]
@artpol84 @hoopoepg The `overlap` test is failing on the Mellanox CI.

<details><Summary>Output from failing `overlap` test in Mellanox CI</summary>
```
00:22:40 + taskset -c 10,11 timeout -s SIGSEGV 17m /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/bin/mpirun -np 4 -bind-to none -mca orte_tmpdir_base /tmp/tmp.j10EKNyIZs --report-state-on-timeout --get-stack-traces --timeout 900 -mca coll '^hcoll' -mca btl_openib_if_include mlx4_0:1 -x UCX_NET_DEVICES=mlx4_0:1 -mca btl_openib_allow_ib true -mca btl self -mca pml yalla /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/thread_tests/thread-tests-1.1/overlap
00:22:40 [1551478960.676925] [jenkins03:25039:0]         sys.c:744  MXM  WARN  Conflicting CPU frequencies detected, using: 3499.98
00:22:40 [1551478960.681230] [jenkins03:25045:0]         sys.c:744  MXM  WARN  Conflicting CPU frequencies detected, using: 3499.98
00:22:40 [1551478960.687256] [jenkins03:25040:0]         sys.c:744  MXM  WARN  Conflicting CPU frequencies detected, using: 3499.98
00:22:40 [1551478960.690290] [jenkins03:25048:0]         sys.c:744  MXM  WARN  Conflicting CPU frequencies detected, using: 3496.92
00:22:41 Time per iteration on each process (ms)
00:22:41 Time 	 Compute time 	 Comm time
00:22:46 [jenkins03:25045:0:25067] address not mapped to object
00:22:46 [jenkins03:25040:0:25065] address not mapped to object
00:22:46 [jenkins03:25048:0:25068] address not mapped to object
00:22:46 [jenkins03:25039:0:25066] address not mapped to object
00:22:46 
==== backtrace ====
00:22:46  0 0x000000000000b976 _dl_relocate_object()  :0
00:22:46  1 0x0000000000013b3c dl_open_worker()  dl-open.c:0
00:22:46  2 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  3 0x00000000000131ab _dl_open()  :0
00:22:46  4 0x0000000000130a02 do_dlopen()  dl-libc.c:0
00:22:46  5 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  6 0x0000000000130ac2 __GI___libc_dlopen_mode()  :0
00:22:46  7 0x000000000000f803 pthread_cancel_init()  :0
00:22:46  8 0x000000000000f9cc _Unwind_ForcedUnwind()  :0
00:22:46  9 0x000000000000dd60 __GI___pthread_unwind()  :0
00:22:46 10 0x0000000000008dd5 __pthread_exit()  :0
00:22:46 11 0x0000000000400d49 threadfunc()  ???:0
00:22:46 12 0x0000000000007dc5 start_thread()  pthread_create.c:0
00:22:46 13 0x00000000000f61cd __clone()  ???:0
00:22:46 ===================
00:22:46 
==== backtrace ====
00:22:46  0 0x000000000000b976 _dl_relocate_object()  :0
00:22:46  1 0x0000000000013b3c dl_open_worker()  dl-open.c:0
00:22:46  2 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  3 0x00000000000131ab _dl_open()  :0
00:22:46  4 0x0000000000130a02 do_dlopen()  dl-libc.c:0
00:22:46  5 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  6 0x0000000000130ac2 __GI___libc_dlopen_mode()  :0
00:22:46  7 0x000000000000f803 pthread_cancel_init()  :0
00:22:46  8 0x000000000000f9cc _Unwind_ForcedUnwind()  :0
00:22:46  9 0x000000000000dd60 __GI___pthread_unwind()  :0
00:22:46 10 0x0000000000008dd5 __pthread_exit()  :0
00:22:46 11 0x0000000000400d49 threadfunc()  ???:0
00:22:46 12 0x0000000000007dc5 start_thread()  pthread_create.c:0
00:22:46 13 0x00000000000f61cd __clone()  ???:0
00:22:46 ===================
00:22:46 
==== backtrace ====
00:22:46  0 0x000000000000b976 _dl_relocate_object()  :0
00:22:46  1 0x0000000000013b3c dl_open_worker()  dl-open.c:0
00:22:46  2 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  3 0x00000000000131ab _dl_open()  :0
00:22:46  4 0x0000000000130a02 do_dlopen()  dl-libc.c:0
00:22:46  5 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  6 0x0000000000130ac2 __GI___libc_dlopen_mode()  :0
00:22:46  7 0x000000000000f803 pthread_cancel_init()  :0
00:22:46  8 0x000000000000f9cc _Unwind_ForcedUnwind()  :0
00:22:46  9 0x000000000000dd60 __GI___pthread_unwind()  :0
00:22:46 10 0x0000000000008dd5 __pthread_exit()  :0
00:22:46 11 0x0000000000400d49 threadfunc()  ???:0
00:22:46 12 0x0000000000007dc5 start_thread()  pthread_create.c:0
00:22:46 13 0x00000000000f61cd __clone()  ???:0
00:22:46 ===================
00:22:46 
==== backtrace ====
00:22:46  0 0x000000000000b976 _dl_relocate_object()  :0
00:22:46  1 0x0000000000013b3c dl_open_worker()  dl-open.c:0
00:22:46  2 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  3 0x00000000000131ab _dl_open()  :0
00:22:46  4 0x0000000000130a02 do_dlopen()  dl-libc.c:0
00:22:46  5 0x000000000000f1b4 _dl_catch_error()  :0
00:22:46  6 0x0000000000130ac2 __GI___libc_dlopen_mode()  :0
00:22:46  7 0x000000000000f803 pthread_cancel_init()  :0
00:22:46  8 0x000000000000f9cc _Unwind_ForcedUnwind()  :0
00:22:46  9 0x000000000000dd60 __GI___pthread_unwind()  :0
00:22:46 10 0x0000000000008dd5 __pthread_exit()  :0
00:22:46 11 0x0000000000400d49 threadfunc()  ???:0
00:22:46 12 0x0000000000007dc5 start_thread()  pthread_create.c:0
00:22:46 13 0x00000000000f61cd __clone()  ???:0
00:22:46 ===================
00:22:47 --------------------------------------------------------------------------
00:22:47 Primary job  terminated normally, but 1 process returned
00:22:47 a non-zero exit code. Per user-direction, the job has been aborted.
00:22:47 --------------------------------------------------------------------------
00:22:47 --------------------------------------------------------------------------
00:22:47 mpirun noticed that process rank 2 with PID 0 on node jenkins03 exited on signal 11 (Segmentation fault).
00:22:47 --------------------------------------------------------------------------
```
</details>

It's segv'ing, which suggests it's timing out.

The problem is clearly not coming from this PR, but my question is: do we have a problem on the v3.1.x branch?  Or is this some kind of local false failure?

**Comment by jsquyres** [comment]
bot:ompi:retest

**Comment by jsquyres** [comment]
bot:mellanox:retest

**Comment by hoopoepg** [comment]
@yosefe look like similar issue with https://github.com/openucx/ucx/issues/3303 - it crashes on jenkins

**Comment by jsquyres** [comment]
bot:mellanox:retest

**Comment by jsquyres** [comment]
Per mail to the core mailing list, the Mellanox CI is having persistent problems right now, and it may take a few days to fix.  This is literally just a README / NEWS / VERSION change, so I'm going to merge.

**Pull request merged**